app/controllers/admin/api_controller.rb

```diff
@@ -5,18 +5,29 @@ class Admin::ApiController < Admin::AdminController
   # If we used "api_key", then our user provider would try to use the value for authentication
 
   def index
-    keys = ApiKey
-      .where(hidden: false)
-      .includes(:user, :api_key_scopes)
+    offset = (params[:offset] || 0).to_i
+    limit = (params[:limit] || 50).to_i.clamp(1, 50)
 
-    # Put active keys first
-    # Sort active keys by created_at, sort revoked keys by revoked_at
-    keys = keys.order(<<~SQL)
-      CASE WHEN revoked_at IS NULL THEN 0 ELSE 1 END,
-      COALESCE(revoked_at, created_at) DESC
-    SQL
+    keys = ApiKey.where(hidden: false)
+    count = keys.count
 
-    render_serialized(keys.to_a, ApiKeySerializer, root: 'keys')
+    keys = keys
+      .includes(:user, :api_key_scopes)
+      # Put active keys first
+      # Sort active keys by created_at, sort revoked keys by revoked_at
+      .order(<<~SQL)
+        CASE WHEN revoked_at IS NULL THEN 0 ELSE 1 END,
+        COALESCE(revoked_at, created_at) DESC
+      SQL
```

Comment:
```suggestion
      .order("revoked_at DESC NULLS FIRST, created_at DESC")
```

I think the ordering above will help PG come up with a better plan without having to scan the whole table.

Reply:
That looks better, thanks!